### PR TITLE
Create go-modules to replace Godeps

### PR DESCRIPTION
**Pull request by apelisse**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,37 @@
+module k8s.io/kube-openapi
+
+go 1.12
+
+require (
+	github.com/NYTimes/gziphandler v0.0.0-20170623195520-56545f4a5d46
```
Comment (liggitt) [inline go.mod]: were these primed from the godeps.json file? should we drop the Godeps dir or do we intend to maintain that file as well?

**Comment by liggitt** [comment]
need to update travis.yml to stop using godeps:
* bump the version of go to 1.12
* drop all the godeps stuff
* set GO111MODULE=on (see https://github.com/kubernetes/client-go/pull/582/files for an example)

File: go.mod (hunk 1 above)
Comment (apelisse) [reply go.mod]: > were these primed from the godeps.json file?

Yes

> should we drop the Godeps dir or do we intend to maintain that file as well?

My gut feeling would be that we should remove it, do you have a recommendation?

**Comment by apelisse** [comment]
Do we need to run some step to verify that the go modules things are up-to-date/correct?

File: go.mod (hunk 1 above)
Comment (liggitt) [reply go.mod]: I'd recommend removing it

**Comment by liggitt** [comment]
> Do we need to run some step to verify that the go modules things are up-to-date/correct?

yes, `go mod tidy` should be zero-diff

File: go.mod (hunk 1 above)
Comment (apelisse) [reply go.mod]: Done

**Pull request merged**